File: Data Structures & Algorithms/daily-temperatures/submission-0.rs
impl Solution {
    pub fn daily_temperatures(temperatures: Vec<i32>) -> Vec<i32> {
        let mut res:Vec<i32> = vec![0; temperatures.len()];
        let mut stack: Vec<usize>= Vec::with_capacity(temperatures.len());
        for i in 0..temperatures.len() {
            while let Some(&prev) = stack.last() {
                if temperatures[prev] >= temperatures[i] {
                    break;
                }
                stack.pop();
                res[prev] = (i - prev) as i32;
            }
            stack.push(i);
        }

        res
    }
}

File: Data Structures & Algorithms/reverse-bits/submission-0.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t res{};
        for(auto i{0}, j{31}; i < 32; ++i, --j) {
            uint32_t bit_to_set{n >> j & 1};
            res |= bit_to_set << i;
        }
        return res;
    }
};
